GiEnJul/ClientApp/src/components/registerGiver/SignUp.tsx

```diff
@@ -1,179 +1,188 @@
-import * as React from 'react';
-import { CssBaseline, Typography, Container } from '@material-ui/core';
-import Confirmation from './Confirmation';
-import ContactInfo from './ContactInfo';
-import LocationGiver from './LocationGiver';
-import SummaryRegistration from './SummaryRegistration';
-import { useEffect, useState } from 'react';
-import getLocations from '../../common/constants/Locations';
+import * as React from "react";
+import { CssBaseline, Typography, Container } from "@material-ui/core";
+import Confirmation from "./Confirmation";
+import ContactInfo from "./ContactInfo";
+import LocationGiver from "./LocationGiver";
+import SummaryRegistration from "./SummaryRegistration";
+import { useEffect, useState } from "react";
+import IGiverFormData from "./IGiverFormData";
+import FamilySizeGiver from "./FamilySizeGiver";
+import getLocations from "../../common/constants/Locations";
+
+const initInputsState: IGiverFormData = {
+  location: "",
+  fullname: "",
+  email: "",
+  phoneNumber: "",
+  maxReceivers: "",
+};
+
+const initState = {
+  step: 1,
+  confirmationOK: false,
+};
 
 const SignUp = () => {
-    const [step, setStep] = useState(1);
-    const [confirmationOK, setConfirmationOK] = useState(false);
+  const [state, setState] = useState(initState);
+  const [inputsState, setInputsState] = useState(initInputsState);
+
+  const [locationOptions, setLocationOptions] = useState<string[]>([]);
+
+  useEffect(() => {
+    getLocations().then((locationArray) => setLocationOptions(locationArray));
+  }, []);
+
+  // go back to previous step
+  const prevStep = (event: React.FormEvent) => {
+    event.preventDefault();
+    setState((prev) => {
```

Comment:
missing () around the state obj, i see that everywhere the setState call was use has the same issue, be careful. Typescript wont tell you its wrong but it can cause bugs.